torch/_inductor/codecache.py

```diff
@@ -615,6 +619,25 @@ def _reduce_unsupported(self, s: Any) -> NoReturn:
         """
         raise BypassFxGraphCache("Reduce unsupported")
 
+    def _reduce_graph_module(
+        self, gm: torch.fx.GraphModule
+    ) -> Tuple[Any, Tuple[Dict[str, Any], str]]:
+        """
+        Custom reducer for graph module to handle irrelevant data for user
+        defined triton kernels
+        Essentially what we are doing here is a huge hack where user defined
+        triton kernel contain a dynamo time side table and the arguments to the
+        call_function are indicies into this side table. These arguments are not
+        for hashing purposes since we included the source code into the cache
+        key and the numbers are prone to give false negatives due to ordering.
+        """
+        fn, (data, imports) = gm.__reduce__()
+        code = data["_code"]
+        code = re.sub(r"kernel_idx = \d+", "", code)
+        code = re.sub(r"constant_args_idx = \d+", "", code)
```

Comment:
maybe these should just both be hashes so we don't need to do the regex delete thing

Reply:
how does fx_graph_cache handle graphs that look similar but have different variable names? I think it's reasonable to have some sort of "graph normalization pass"

Reply:
it does not, unless the graph are identical, we cache miss. I thought this was the safest starting point. We could do some normalization eventually but i want to understand how much does this bring us